### PR TITLE
fix #175216, fix #269114: Crash with shortcut alt+arrow up/down if mo…

### PR DESCRIPTION
**Pull request by mattmcclinch**
…re than one element or nothing is selected

Fix other related issues with the up-chord/down-chord commands
Prevent crash when adjusting the grips of ties

### CODE REVIEW

File: libmscore/tie.cpp [hunk 1]
```diff
@@ -153,10 +153,9 @@ void TieSegment::editDrag(EditData& ed)
             //
             if ((g == Grip::START && isSingleBeginType()) || (g == Grip::END && isSingleEndType())) {
                   Spanner* spanner = tie();
-                  Note* note = toNote(ed.view->elementNear(ed.pos));
```
Comment (mattmcclinch) [inline libmscore/tie.cpp]: Casting to a Note before checking that the Element is a Note causes a crash.

File: mscore/scoreview.cpp [hunk 2]
```diff
@@ -1904,10 +1904,15 @@ void ScoreView::cmd(const char* s)
             }
       else if (cmd == "up-chord") {
             Element* el = score()->selection().element();
+            Element* oel = el;
             if (el && (el->isNote() || el->isRest()))
                   cmdGotoElement(score()->upAlt(el));
             el = score()->selection().element();
-            while (el->isRest() && toRest(el)->isGap() && el->voice() != 0) {
```
Comment (mattmcclinch) [inline mscore/scoreview.cpp]: Dereferencing _el_ before making sure that it is not NULL causes a crash. Also, gaps in voice 1 should not be possible. But if there were a gap in voice 1, still handle it correctly.

File: mscore/scoreview.cpp [hunk 3]
```diff
@@ -1918,7 +1923,7 @@ void ScoreView::cmd(const char* s)
             if (el && (el->isNote() || el->isRest()))
                   cmdGotoElement(score()->downAlt(el));
             el = score()->selection().element();
-            while (el->isRest() && toRest(el)->isGap() && el->voice() != 3) {
```
Comment (mattmcclinch) [inline mscore/scoreview.cpp]: Making sure that `el->voice() != 3` defeats the purpose of the following few lines, which correctly terminate the _while_ loop.

File: libmscore/navigate.cpp [hunk 4]
```diff
@@ -197,10 +197,9 @@ Element* Score::upAlt(Element* element)
             Chord* chord = note->chord();
             const std::vector<Note*>& notes = chord->notes();
             auto i = std::find(notes.begin(), notes.end(), note);
-            if (i != notes.begin()) {
```
Comment (mattmcclinch) [inline libmscore/navigate.cpp]: This causes Alt+up to do nothing if the bottom note of the chord is selected.

File: libmscore/tie.cpp [hunk 5]
```diff
@@ -153,10 +153,9 @@ void TieSegment::editDrag(EditData& ed)
...
-                  if (note && note->isNote()
-                     && ((g == Grip::END && note->tick() > tie()->tick()) || (g == Grip::START && note->tick() < tie()->tick2()))
-                     ) {
+                  Element* e = ed.view->elementNear(ed.pos);
+                  Note* note = (e && e->isNote()) ? toNote(e) : NULL;
```
Comment (lasconic) [inline libmscore/tie.cpp]: We prefer nullptr than NULL.

**Comment by lasconic** [comment]
Fixed the NULL -> nullptr and merged manually in 2e112abcff53d72ac4c4a3823afcef77b5e20403

**Pull request closed without merging**